### DISCUSSION TITLE
StartTimeUnixNano is always set to "11651379494838206464" using v0.37.0 version of go.opentelemetry.io/otel/metric

### DISCUSSION THREAD

**Discussion by anil181** [open-telemetry/opentelemetry-go#4265] [Q&A]
I am using Int64ObservableGauge instrument with callback (RegisterCallback) to record a metric with NewPeriodicReader and export it to otel collector. Below is the code on how am doing it:
```
			observable, _ := meter.Int64ObservableGauge(
				fullName,
				instrument.WithDescription(metricDescription),
			)


			//callback function can be updated periodically w/o necessity to create new instance of instrument
			registration, _ := meter.RegisterCallback(func(_ context.Context, o metric.Observer) error {
				if bObserve {
					o.ObserveInt64(observable, metricValue, statsLabels...)
					bObserve = false
				}
				return nil
			}, observable)
```
But with all the metric being sent i see the startTimeUnixNano is always set to "11651379494838206464", however the timeUnixNano is set properly.

excerpt from the oltp.json
```
                     "gauge":{
                        "dataPoints":[
                           {
                              "attributes":[
                                 {
                                    "key":"applicationName",
                                    "value":{
                                       "stringValue":"imperative:application"
                                    }
                                 },
                                 {
                                    "key":"metricName",
                                    "value":{
                                       "stringValue":"clientside.max_conns"
                                    }
                                 },
                                 {
                                    "key":"metricProperty",
                                    "value":{
                                       "stringValue":"/app/imperative:application/stack/imperative:stack"
                                    }
                                 },
                                 {
                                    "key":"metricType",
                                    "value":{
                                       "stringValue":"my_server_stat"
                                    }
                                 },
                                 {
                                    "key":"stackName",
                                    "value":{
                                       "stringValue":"imperative:stack"
                                    }
                                 },
                                 {
                                    "key":"systemId",
                                    "value":{
                                       "stringValue":"edd50c42-c08c-e2bb-bcb6-a912e090ecd0"
                                    }
                                 }
                              ],
                              "startTimeUnixNano":"11651379494838206464",
                              "timeUnixNano":"1685016941030560013",
                              "asInt":"0"
                           }
                        ]
                     }
```
Earlier we were using meter.AsyncInt64().Gauge from v0.31.0 version of go.opentelemetry.io/otel/metric and the startTimeUnixNano was populated with the same value as that of timeUnixNano.

I could not find a way to set the value of startTimeUnixNano, could you please guide on fixing this?

Thanks & regards,
Anil

**Comment by pellared** [comment]
The values are set by the SDK. Why do you think the values are wrong? Please see: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#gauge

**Reply by anil181** [reply]
@pellared Hi Robert, i am not sure why the values are wrong. But the startTimeUnixNano should have been closer to the timeUnixNano. The application is running inside container in Kubernetes environment where the system time is set properly. 

Is this by any chance a bug in the SDK?

**Reply by pellared** [reply]
> (optional) A timestamp (start_time_unix_nano) which best represents the first possible moment a measurement could be recorded. This is commonly set to the timestamp when a metric collection system started.

Is the value a lot different than the moment the application has started?

The value should not change during a distinct process (SDK) execution.

**Reply by anil181** [reply]
Yes, the value "11651379494838206464" is a future date "Tuesday, March 21, 2339 10:18:14.838 PM" when converted to "human-readable date" and the value remains the same across SDK executions.
<img width="1201" alt="Screenshot 2023-06-26 at 20 44 20" src="https://github.com/open-telemetry/opentelemetry-go/assets/7187168/d8097ce4-20b2-495d-af75-458638f958ff">

**Reply by pellared** [reply]
The value is indeed strange. Recently I got a valid value: https://github.com/open-telemetry/opentelemetry-go/issues/4133#issuecomment-1568713968

Can you please:

1. Try using the latest version.
 
If the bug persists then:

2. Create a bug with [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)

**Reply by anil181** [reply]
The difference between my and your example is that am using asynchronous instrument Int64ObservableGauge while in https://github.com/open-telemetry/opentelemetry-go/issues/4133#issuecomment-1568713968 you are using synchronous Float64UpDownCounter. Have you tried with asynchronous instrument?

**Reply by pellared** [reply]
I have managed to reproduce it and I agree that this is a bug. I will create an issue as I already have some repro steps. I also found places in code that should be investigated while fixing.

**Reply by pellared** [reply]
I created https://github.com/open-telemetry/opentelemetry-go/issues/4268 and https://github.com/open-telemetry/opentelemetry-go/issues/4269

> could you please guide on fixing this?

I started working on https://github.com/open-telemetry/opentelemetry-go/issues/4268

Regarding https://github.com/open-telemetry/opentelemetry-go/issues/4269 you can take a look and consider contributing 😉

**Reply by anil181** [reply]
Thank you @pellared for reproducing the issue on your end and proposing the fix. For https://github.com/open-telemetry/opentelemetry-go/issues/4269 _"A timestamp (start_time_unix_nano) which best represents the first possible moment a measurement could be recorded. This is commonly set to the timestamp when a metric collection system started."_ the statement is not lucid and i see you following up on the clarification.
